controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/PolledBuildSystem.java

```diff
@@ -77,7 +81,14 @@ public void removeJobs(ApplicationId application) {
                 Deque<ApplicationId> queue = curator.readJobQueue(jobType);
                 for (ApplicationId a : queue) {
                     ApplicationId application = removeFromQueue ? queue.poll() : a;
-                    jobsToRun.add(new BuildJob(projectIdFor(application), jobType.id()));
+
+                    Optional<Long> projectId = projectId(application);
+                    if (projectId.isPresent()) {
+                        jobsToRun.add(new BuildJob(projectId.get(), jobType.id()));
+                    } else {
+                        log.warning("Not queuing " + jobType.id() + " for " + application.toShortString() +
```

Comment:
Won't this cause WARNING-spam in CD, if all integration tests end up triggering this?

Reply:
No, because they are self-triggering. Basically, this only happens for applications that are not self-triggering and is missing project ID, and if that happens I want some noise in the logs. :-)

Reply:
Sounds perfectly reasonable. Question answered :]